velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -107,9 +107,9 @@ class Addition {
 #endif
 #endif
   {
-    r.setUnscaledValue(
-        a.unscaledValue() * DecimalUtil::kPowersOfTen[aRescale] +
-        b.unscaledValue() * DecimalUtil::kPowersOfTen[bRescale]);
+    r = checkedPlus<R>(
+        checkedMultiply<R>(R(a), R(DecimalUtil::kPowersOfTen[aRescale])),
```

Comment:
Do we need to use checkedMultiply here? What would be a scenario where it fails?

Reply:
eg: Adding UnscaledLongDecimal::max() (which is 38 digits long) and  0.01, would require shifting max() value by 2 places. This is an overflow case.
```

99999999999999999999999999999999999999 + 0.01
After rescaling LHS:
= 9999999999999999999999999999999999999900 + 1
```

Reply:
I understand why we need checkedPlus, but why do we need checkedMultiply. In which case checkedMultiply fails? Is there a test case?

Reply:
In the above example of `99999999999999999999999999999999999999 + 0.01`.

99999999999999999999999999999999999999 (LHS) will be multiplied by 100 (rf) through checkedMultiply<UnscaledLongDecimal>. This will overflow.

I am not particularly testing rescaling failure in addition case. The Decimal multiply tests which also use `checkedMultiply<UnscaledLongDecimal>` call check for overflow failures.

I have added test case under add for this failure condition.

Reply:
I think Masha's point is to add tests for all the possible overflow cases for addition, and multiply functions. 
Addition uses a `checkedPlus` and a `checkedMultiply` (for rescale). I see that you added a test for each.
Can you do the same for multiply also? There are two `checkedMultiply` uses and both can overflow.
Can you add a test and the comment explaining which function overflows?
The tests will help us understand why we need `checked` functions in each case.